### PR TITLE
OHRI-1226 patient list widget enhancements

### PR DESCRIPTION
**Pull request by pirupius**
This has the first version of the newly improved patient list widget with a data table.

- Pending work includes fixing some styles to match designs

### CODE REVIEW

File: packages/esm-ohri-core-app/src/ohri-dashboard/home/ohri-home.component.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import styles from './ohri-home.scss';
+import { useTranslation } from 'react-i18next';
+import { PatientList, PatientListDataTable } from '@ohri/openmrs-esm-ohri-commons-lib';
+
+const OhriHome: React.FC = () => {
+  const { t } = useTranslation();
+  return (
+    <div className={styles.ohriHomeContainer}>
+      {/* <PatientList /> */}
```
Comment (samuelmale) [inline packages/esm-ohri-core-app/src/ohri-dashboard/home/ohri-home.component.tsx:10]: @pirupius Can we get rid of this and the component if it's not being used?

File: packages/esm-ohri-core-app/tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "extends": "../../tsconfig.json",
-  "include": ["src/**/*"],
+  "include": ["src/**/*", "../esm-commons-lib/src/components/patient-lists/patient-list.component.tsx", "../esm-commons-lib/src/components/patient-lists/patient-list.component.tsx"],
```
Comment (samuelmale) [inline packages/esm-ohri-core-app/tsconfig.json:3]: Why do you need this? I think this config needs to go to the target package: https://github.com/UCSF-IGHS/openmrs-esm-ohri/blob/master/packages/esm-commons-lib/tsconfig.json

**Review by samuelmale — CHANGES_REQUESTED** [review]
LGTM, just a few comments.

Can we add pagination to this new nice component?

File: packages/esm-ohri-core-app/tsconfig.json (hunk 2 above)
Comment (pirupius) [reply packages/esm-ohri-core-app/tsconfig.json:3]: Let me revert this

**Comment by pirupius** [comment]
@samuelmale i've made the changes to the 2 comments. This widget is still work in progress so some other features like pagination will follow with some other UI tweaks.

**Comment by samuelmale** [comment]
>This widget is still work in progress so some other features like pagination will follow with some other UI tweaks.

Do we have a ticket to keep track of the above?

**Comment by pirupius** [comment]
Let me create them

File: packages/esm-commons-lib/src/api/api.ts [hunk 3]
```diff
@@ -9,6 +9,8 @@ import {
   covid_Assessment_EncounterUUID,
   covidOutcomesCohortUUID,
 } from '../constants';
+import { FhirPatient, FhirPatientResponse, PatientListRow } from './types';
+import useSWR from 'swr';
```
Comment (samuelmale) [inline packages/esm-commons-lib/src/api/api.ts:13]: Do you need this?

File: packages/esm-ohri-core-app/src/index.ts [hunk 4]
```diff
@@ -51,9 +51,9 @@ function setupOpenMRS() {
         offline: true,
       },
       {
-        id: 'ohri-all-patients-list',
+        id: 'ohri-home',
         slot: 'ohri-home-dashboard-slot',
-        load: getAsyncLifecycle(() => import('./components/all-patients-list/patient-list.component'), {
+        load: getAsyncLifecycle(() => import('./ohri-dashboard/home/ohri-home.component'), {
```
Comment (samuelmale) [inline packages/esm-ohri-core-app/src/index.ts:56]: What you are actually doing here is: Plugging the "OhriHome" component into the "Ohri Home Dashboard" which introduces some confusion to the tree. You only want to plug the patient list into the "Ohri Home Dashboard". Simply rename the "OhriHome" component at path: `/ohri-dashboard/home/ohri-home.component` to something like "AllPatientsList"

**Review by samuelmale — CHANGES_REQUESTED** [review]
Almost there

File: packages/esm-commons-lib/src/api/api.ts (hunk 3 above)
Comment (pirupius) [reply packages/esm-commons-lib/src/api/api.ts:13]: This is residue after moving the new endpoint to another file and need to be removed 👍🏾

File: packages/esm-ohri-core-app/src/index.ts (hunk 4 above)
Comment (pirupius) [reply packages/esm-ohri-core-app/src/index.ts:56]: That's okay...let me undo this and use the previous setup

File: packages/esm-commons-lib/src/components/patient-lists/patient-list.component.tsx [hunk 5]
```diff
@@ -92,7 +88,7 @@ const PatientList: React.FC<PatientListProps> = () => {
   const getPatientURL = (patientUuid) => `/openmrs/spa/patient/${patientUuid}/chart`;
 
   return (
-    <>
+    <div className={styles.patientListContainer}>
```
Comment (samuelmale) [inline packages/esm-commons-lib/src/components/patient-lists/patient-list.component.tsx:91]: Isn't this the old component? If so, I assumed we were dropping it.

Comment (samuelmale) [reply packages/esm-commons-lib/src/components/patient-lists/patient-list.component.tsx]: @pirupius did you see my comment above?

**Review by samuelmale — APPROVED** [review]

**Pull request merged**